src/store/mod.rs

```diff
@@ -5,36 +5,35 @@ pub mod s3store;
 pub mod zdb;
 
 use rand::seq::SliceRandom;
-use std::{collections::HashMap, pin::Pin};
 
 pub use bs::BlockStore;
-use futures::Future;
 
-lazy_static::lazy_static! {
-    static ref STORES: HashMap<String, Factory> = register_stores();
-}
-
-/// register_stores is used to register the stores built in types
-/// so they can be created with a url
-fn register_stores() -> HashMap<String, Factory> {
-    let mut m: HashMap<String, Factory> = HashMap::default();
-    m.insert("dir".into(), dir::make);
-    m.insert("zdb".into(), zdb::make);
-    m.insert("s3".into(), s3store::make);
-    m.insert("s3s".into(), s3store::make);
-    m.insert("s3s+tls".into(), s3store::make);
-
-    m
-}
-
-pub async fn make<U: AsRef<str>>(u: U) -> Result<Box<dyn Store>> {
+pub async fn make<U: AsRef<str>>(u: U) -> Result<Stores> {
     let parsed = url::Url::parse(u.as_ref())?;
-    let factory = match STORES.get(parsed.scheme()) {
-        None => return Err(Error::UnknownStore(parsed.scheme().into())),
-        Some(factory) => factory,
-    };
 
-    factory(u.as_ref()).await
+    if parsed.scheme() == dir::SCHEME {
+        return Ok(Stores::Dir(
+            dir::DirStore::make(&u)
+                .await
+                .expect("failed to make dir store"),
+        ));
+    }
+    if parsed.scheme() == "s3" {
```

Comment:
this should also be done for `s3s` and `s3s+tls` as per original map

i think also you can use a match instead of multiple `if` statements